ml_genn/model.py

```diff
@@ -290,108 +303,229 @@ def convert_tf_model(tf_model, converter=Simple(),
-                    layer = Conv2D(
-                        name=tf_layer.name, filters=tf_layer.filters,
-                        conv_size=tf_layer.kernel_size,
-                        conv_strides=tf_layer.strides,
-                        conv_padding=tf_layer.padding,
-                        connectivity_type=connectivity_type, 
-                        neurons=converter.create_neurons(tf_layer, pre_compile_output))
+
+            # find inbound layers
+            tf_in_layers = []
+            for in_layer in [n.inbound_layers for n in tf_layer.inbound_nodes
+                             if n in tf_model_nodes]:
+                if isinstance(in_layer, list):
+                    tf_in_layers += in_layer
                 else:
-                    print('converting AveragePooling2D -> Conv2D layers <{}>'.format(tf_layer.name))
-                    layer = AvePool2DConv2D(
-                        name=tf_layer.name, filters=tf_layer.filters,
-                        pool_size=pool_layer.pool_size, conv_size=tf_layer.kernel_size,
-                        pool_strides=pool_layer.strides, conv_strides=tf_layer.strides,
-                        pool_padding=pool_layer.padding, conv_padding=tf_layer.padding,
-                        connectivity_type=connectivity_type, 
-                        neurons=converter.create_neurons(tf_layer, pre_compile_output))
+                    tf_in_layers.append(in_layer)
+            tf_in_layers_all[tf_layer] = tf_in_layers
 
-                layer.connect([previous_layer])
-                layer.set_weights(tf_layer.get_weights())
+            # find outbound layers
+            tf_out_layers = [n.outbound_layer for n in tf_layer.outbound_nodes
+                             if n in tf_model_nodes]
+            tf_out_layers_all[tf_layer] = tf_out_layers
 
-                model.layers.append(layer)
-                previous_layer = layer
-                pool_layer = None
 
-            # === AveragePooling2D Layers ===
-            elif isinstance(tf_layer, tf.keras.layers.AveragePooling2D):
-                print('deferring AveragePooling2D layer <{}>'.format(tf_layer.name))
+        # Perform any pre-compilation tasks
+        pre_compile_output = converter.pre_compile(tf_model)
 
-                pool_layer = tf_layer
+        # configure ML GeNN model build process
+        mlg_model_inputs = []
+        mlg_model_outputs = []
+        mlg_layer_lookup = {}
+        new_tf_layers = set()
+        traversed_tf_layers = set()
+
+        if isinstance(tf_model, tf.keras.models.Sequential):
+            tf_in_layers = tf_in_layers_all[tf_model.layers[0]]
+            assert(len(tf_in_layers) == 1)
+            tf_out_layers = [n.outbound_layer for n in tf_in_layers[0].outbound_nodes
+                             if n in tf_model_nodes]
+            tf_out_layers_all[tf_in_layers[0]] = tf_out_layers
+
+        else:
+            tf_in_layers = [tf_model.get_layer(name) for name in tf_model.input_names]
```

Comment:
No, because Sequential models get rid of their input layer object. See below.

Reply:
Ahh I see - ``input_names`` has it's name but for whatever reason it can't be extracted - bizarre. Maybe add a  comment about why you handle the two differently

Reply:
Allright, will do.